ufl/algorithms/compute_form_data.py

```diff
@@ -348,7 +346,31 @@ def compute_form_data(form,
     self.reduced_coefficients = sorted(reduced_coefficients_set,
                                        key=lambda c: c.count())
     self.num_coefficients = len(self.reduced_coefficients)
-    self.original_coefficient_positions = [i for i, c in enumerate(self.original_form.coefficients())
+
+    # Store all the external operators and their derivative multiindex
+    # turning up in the form (Differentiation may have introduced new external operators)
+    extops_pos = {e._extop_master.count(): i for i, e in reversed(list(enumerate(self.original_form.external_operators())))}
```

Comment:
Given you're putting things in a dict, why not just `{e._extop_mast.count(): i for i, e in enumerate(...)}`?

Reply:
Because otherwise we end up with the wrong position (it is not a one-to-one relation). Let’s say that we start from an external operator `p(u)` and its derivative `dpdu(u)`, because external operators are sorted respectively with their count value `form.external_operators()` will be [`p`, `dpdu`] as `dpdu` was made after `p` (dpdu.count() > p.count()).  Now to refer to the derivatives of the external operator `p` we store the position of `p`, so `enumerate(...)` does not do the job as it will consider the position of `dpdu` as `dpdu._extop_master = p` and `p._extop_master = p`.